File: Modulo14/Funcionario-Java/Medico.java

public class Medico extends Funcionario {

    private String especialidade;

    public Medico(int CPF, String nome, String especialidade) {
        super(CPF, nome);
        this.especialidade = especialidade;
    }

    public String getProfissao() {
        return "Medico";
    }

    public String getEspecialidade() {
        return especialidade;
    }

    public void imprime() {
        super.imprime();
        System.out.println("Especialidade: " + especialidade);
    }
}

File: Modulo16/Funcionarios/Medico.cpp
#include "Medico.h"

Medico::Medico(int CPF, string nome, double salario, string especialidade) :
Funcionario(CPF, nome, salario), especialidade(especialidade) {
}

Medico::~Medico() {
    cout << "Destrutor Medico (" << this->getCPF() << ") invocado" << endl;
}

int Medico::getProfissao() const {
    return MEDICO;
}

string Medico::getEspecialidade() const {
    return especialidade;
}

void Medico::imprime() const {
    Funcionario::imprime();
    cout << "Especialidade : " << especialidade << endl;
}



